go/README.md

```diff
@@ -1 +1,200 @@
+
+Don't use a name that is meaningless, like: util, common, misc.
+There is one exception to this: helpers.
+We will use helpers to store various helper-functions and over time, we can decide to group them into
+a package that is more suitable, of move them to Capila or Libgoaz.
+
+## Imports
+
+Order of imports:
+- Internal go packages
+- External packages (3rd party)
+- Aan Zee library packages
+- Aan Zee packages of current project
+
+## Variable declaration
+
+Top level order of declaration:
+- Constants
+- Types
+- Variables
+
+Global / package / function
+
+No Global Variables. Use getters for that.
+
+If a package variable or function is used in more than 1 file, put it in a base.go or a file starting with base.
+
+Declare a function variable at the top of the function, or close to the first usage.
+
+## Functions and methods
+
+You can house more than 1 function inside a file (inside a package).
+The file name represents the content it houses.
+
+- Naming
+	- Give the function a name it represent
+	- Don't abbreviate
+	- List / Get / Create / Update / Delete
+
+- Return variables:
+  - When returning an error, it is always the last value that returns
+
+## Testing
+
+- Code coverage (100%, unless it's a project decision not to do so)
+- No '.' imports
+
+- Fuzzy testing on high level
+- Handlers should only test the basic working
+- Services should test every scenario possible and should be tested with 100% codecoverage
+  - If there are model validations, these should be tested too
+- Repositories should test every scenario possible and should be tested with 100% codecoverage
+
+## Comments
+
+- Package
+  - Doc
+    - Describes why the package exists
+    - It has the package format:
+      - Package the_name the_rest_of_the_text
+- Variable
+  - Single line
+    - Format of that:
+    // Explain what it does and/or why it exists
+
+- Function/method
+  - Multi line / single line
+    - Is written in third-person singular with minimal 4 words
+     - Format of that:
+	/*
+	YourFunction does something spectacular.
+	*/
+	or:
+	// YourFunction does something spectacular.
+
+## Variable naming
+
+A variable name is always named after the thing it contains.
+
+In a loop, like in this example:
+for userIndex,userRecord := range userRecords {
+	println("userIndex:", userIndex)
+	println("userRecord:", userRecord)
+}
+
+If it can hold a collection of things, name it plural, like "users".
+If it can only hold one value, name it singular, like "user".
+if it is a boolean var, start it with something like "has", "should", "would" ,"can", "could" or "is".
+
+It's also common use, when a variable name holds an abbreviation, that the letters are all capitals, or all lowercase.
+like "SQLStatement" or "sqlStatement"
+
+## Linting
+
+It is nice to start a linter now and then.
+Make check, or make check2 will run the linters and will check your files for "mistakes".
+Try to improve your code with the hints you get from them.
```

Comment:
Lint rules should be core. Without the correct 'style', the linter should fail with error code, thus the pull request cannot be merged